cmd/jaeger/internal/integration/span_reader.go

```diff
@@ -0,0 +1,172 @@
+// Copyright (c) 2024 The Jaeger Authors.
+// SPDX-License-Identifier: Apache-2.0
+
+package integration
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"io"
+	"math"
+	"strings"
+	"time"
+
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials/insecure"
+	"google.golang.org/grpc/status"
+
+	"github.com/jaegertracing/jaeger/model"
+	"github.com/jaegertracing/jaeger/ports"
+	"github.com/jaegertracing/jaeger/proto-gen/api_v2"
+	"github.com/jaegertracing/jaeger/storage/spanstore"
+)
+
+var (
+	_ spanstore.Reader = (*spanReader)(nil)
+	_ io.Closer        = (*spanReader)(nil)
+)
+
+// SpanReader retrieve span data from Jaeger-v2 query with api_v2.QueryServiceClient.
+type spanReader struct {
+	Port int
+
+	clientConn *grpc.ClientConn
+	client     api_v2.QueryServiceClient
+}
+
+func createSpanReader(port int) *spanReader {
+	return &spanReader{
+		Port: port,
+	}
+}
+
+func (r *spanReader) Start() error {
```

Comment:
I suggest merging this into create function. What's the benefit of separating Start?

Reply:
Okay. I've seen a lot of structs that separate the creation and start functions so I thought it was cleaner that way. I'll refactor this.